datareservoirio/client.py

```diff
@@ -280,16 +278,29 @@ def get(
         if start >= end:
             raise ValueError("start must be before end")
 
-        time_start = timeit.default_timer()
-
-        log.debug("Getting series range")
-        series = (
-            self._storage.get(series_id, start, end)
-            .set_index("index")
-            .squeeze("columns")
-            .loc[start:end]
-            .copy(deep=True)
+        # make an "expolotary" request
+        response = self._auth_session.get(
+            environment.api_base_url
+            + f"timeseries/{series_id}/data/days?start={start}&end={end}"
         )
+        response.raise_for_status()
+        response_json = response.json()
+
+        if response_json:
+            with ThreadPoolExecutor(max_workers=None) as e:
+                futures = [
+                    e.submit(
+                        self._storage.get,
+                        environment.api_base_url
+                        + f"timeseries/{series_id}/data/days?start={val_i}&end={val_i}",
+                    )
+                    for val_i in _timeseries_available_days(response_json)
+                ]
+            df = pd.concat([future_i.result() for future_i in futures])
+        else:
+            df = pd.DataFrame(columns=("index", "values")).astype({"index": "int64"})
```

Comment:
This will fail when no data is available, since `response_json = {"Files": []}` (which is not `False`).

Reply:
You are right. I'll fix that. Maybe it should be
```
if response_json["Files"]:
```

Reply:
Yes, or something like this?
`if days := _timeseries_available_days(response_json)`